README.md

```diff
@@ -21,147 +24,131 @@ or to use in an npm script
 
     $ npm run nodeshift
 
-### Example Usage
-
-#### Prerequisites
+## Core Concepts
 
-* Minishift
+### Commands & Goals
 
-For this example, we will run against an Openshift Cluster running locally using Minishift.
+By default, if you run just `nodeshift`, it will run the `deploy` goal, which runs everything.
 
-There are directions here to install [Minishift](https://docs.openshift.org/latest/minishift/getting-started/installing.html)
+**resource** - will parse and create the application resources files on disk
 
-_I installed using Homebrew and use Virtual Box as my hypervisor provider, so my command to initially start minishift was something like this: `minishift start --vm-driver=virtualbox`_
+**apply-resource** - does the resource goal and will also push that to your running cluster
 
+**build** - archives the code, creates build config and imagestream and pushes the binary to the cluster
 
-* OC command line tools
+**deploy** - does everything, the archiving of code and everything that apply-resource does
 
-These are also part of the minishift install, and [step 2 of this section](https://docs.openshift.org/latest/minishift/getting-started/quickstart.html#starting-minishift) tells you how to put them on your path.
+**undeploy** - removes things that were deployed with the apply-resource command
 
-This guide also assumes you have a modern version(4+) of Node.js and npm installed.
 
-#### Example
+### .Nodeshift Directory
```

Comment:
It's not totally clear from this description that it's the user's responsibility to supply the resource files in the `.nodeshift` directory. What is the minimum set of resource files required for a project?